modules/data-entry/src/main/frontend/src/questionnaire/PedigreeQuestion.jsx

```diff
@@ -127,7 +127,7 @@ function PedigreeQuestion(props) {
   var onSetAnswerPath = function (answerPath) {
     const entry = /\/Forms\/([^.]+)/.exec(location.pathname);
     if (entry) {
-      return setAnswerPath(entry[0] + answerPath.replace(props.path, ''));
+      return setAnswerPath(entry[0] + answerPath.replace(/^\./, ''));
```

Comment:
In `FileResourceQuestion` this is done as:

```suggestion
      return setAnswerPath(reader["/URL"] + "/" + answerPath);
```

(Didn't test it)

In `FileResourceQuestion`, this is done not in a callback, but in the normal code flow of the component's rendering. I don't like that we have two different implementations of the same mechanism, should we try to unify them?

Also, not sure why we need a dedicated delete request for this, isn't it enough to just empty the values for the answer node, and let it be processed like any regular answer?

Reply:
I believe the reason was to be able to reuse the `DeleteButton` with its confirmation functionality. However, since the button was slightly modified in commit ccf5bd35, saving the form and delete requests no longer seem necessary. I rebased on `dev` to include that commit, and refactored in commit fa20bd6. Please re-evaluate.